### ISSUE TITLE
Unable to enter in recovery mode

### ISSUE THREAD

**Issue by Yuvrajaryans** [TeamWin/Team-Win-Recovery-Project#992]
Hello,
         I have rooted phone, unlocked bootloader then flash recovery image using twrp app. It shows flashed successfully but, I m unable enter in recovery mode. Tried with hard key, from twrp app and from other apps like quick reboot but all failed to enter in recovery mode.  
Plz help me..

**Comment by JonnyTech** [comment]
Try this from bootloader:

```
fastboot flash RECOVERY recovery.img
fastboot boot recovery.img
```

Replace **RECOVERY** and **recovery.img** with those appropriate to you.

If the phone boots to Android then you have to try again.

**Comment by mhedoc** [comment]
The last TWRP with version number 3.1.1.1 may be broken. It's only 28.6M in size, flashing it with previous TWRP will break it, flashing it in fastboot also. Flashing previous version(3.1.1.0) is working. Or do i miss something, do we have to update in special way? Thank you.

Device is Leeco Le Max 2 (x820 aka x2)
Best regards

**Comment by Yuvrajaryans** [comment]
it didnt flash . it shows download recovery is not of proper size. (10.43mb).
so can u plz send exact recovery image for my device sony experia z 6603 (yuga).
i had a doubt because i have search in other site and there it is of 16 mb. recovery image for yuga.
so plz help.

**Comment by Yuvrajaryans** [comment]
![image](https://user-images.githubusercontent.com/29604128/27763262-d9dd9b40-5e9d-11e7-932e-8942cd9c9c1c.png)
it is showing this error .
plz help me .

**Comment by Yuvrajaryans** [comment]
Plz help me. 
Pllzzzzzz

**Comment by JonnyTech** [comment]
I suspect that you have not unlocked your bootloader correctly.
https://www.skyneel.com/unlock-bootloader-letv-leeco-le-max-2
fastboot -i 0x2A96 oem unlock

**Comment by JonnyTech** [comment]
Ah, I notice two different posters in this thread, the above was for @mhedoc

**Comment by JonnyTech** [comment]
@Yuvrajaryans did you unlock using a key as detailed here?: https://forum.xda-developers.com/xperia-z/development/tutorial-xperia-z-zl-bootloader-t2153261

**Comment by Yuvrajaryans** [comment]
Yes I have unlocked from the instruction given in Sony developer website.

**Comment by JonnyTech** [comment]
The message indicated that your bootloader is **not** unlocked. Try unlocking it again.

**Comment by Yuvrajaryans** [comment]
Is there any problem in rooting also because I have rooted phone by just installing king root app.

**Comment by JonnyTech** [comment]
Once you have TWRP installed and working then rooting should be as simple as flashing SuperSU from recovery.

**Comment by Yuvrajaryans** [comment]
Ok then I will relock and then unlock it again.

**Comment by Yuvrajaryans** [comment]
I have tried unlocking from every available option but still it is showing same error. 
Plz help

**Comment by Yuvrajaryans** [comment]
One more thing to clear and that is when just started this flashing stuff I was struck in twrp recovery mode because I had flash recovery image in boot partation,  so I have flashed Sony firmware 10.7.A.0.228 using flash tool. Plz help.

**Comment by JonnyTech** [comment]
Please provide a transcript of the unlocking and flashing session (copy and paste from the terminal). You are missing some vital step but it is difficult to determine without logs.

**Comment by Yuvrajaryans** [comment]
sorry, I was out of station.
For bootloader unlocking.
1. Turning off Xperia™ Z.
2. Connect a USB-cable to computer.
4. When device is connected, opened command window on my computer and opened platform-tools folder within the Android SDK folder.
5. Entered the following command:
>fastboot devices
C. Entered unlock key
>fastboot -i 0x0fce oem unlock 0x1C2192CC70C50744
2.. Done! checked in device by entering number *#*#7378423#*#*
showing in bootloader unlocked "yes".
  
For Flashing Recovery.
1. downloaded twrp recovery for my device.
    placed that recovery img. in platform tool folder of sdk folder.
2. opened command window there .
3. connecting device with computer.
4. entering command in command window "adb reboot bootloader." device boots into bootloader indicating blue light. (some time it didnt enter in bootloader indicating device not connected but then after entering command again it enter in bootloader)
5. Then again enter command "fastboot flash recovery twrp"
after that error which i have placed earlier was shown.
plz help me .   plzzzzzzzzzzzzzz.

**Comment by Yuvrajaryans** [comment]
Plz reply.....

**Comment by JonnyTech** [comment]
What is the output of these commands from bootloader:
```
fastboot devices
fastboot getvar all
```

**Comment by JonnyTech** [comment]
Also in phone dialler type in `*#*#7378423#*#*`
Go to `Service Info > Configuration` and check the Rooting Status
What is the value of Bootloader lock?

**Comment by Yuvrajaryans** [comment]
Status is showing yes
Bootloader unlocked: yes

**Comment by nailyk-fr** [comment]
its a xperia. AFAIK Z don't have bootloader with recovery access. Your kernel need to support extract_elf_ramdisk to be able to boot FOTAKernel recovery (all AOSP based rom will). 
If your using stock it depend of your rom.

**Comment by Yuvrajaryans** [comment]
plz escalate....

**Comment by Yuvrajaryans** [comment]
THIS THE DETAILS OF KERNEL VERSION. 
Android version : 5.1.1 / kernel version : 3.4.0-perf-gbccb33a / Build number : 10.7.A.0.222

**Comment by nailyk-fr** [comment]
Ask your rom maintainer or re read your xda thread for how to install/access recovery. 
Rom stock (mostly) wont support FOTAKernel recoveries. 
You need a kernel who support _extract_elf_ramdisk_.

**Comment by Yuvrajaryans** [comment]
This is the output of the comand: 
fastboot devices
fastboot getvar all    is 

![image](https://user-images.githubusercontent.com/29604128/28375349-de2b2c1e-6cc4-11e7-9438-440d7a16972e.png)

**Comment by JonnyTech** [comment]
Very strange that getvars is empty. It looks like the fastboot access from the bootloader is incomplete. The advice from @nailyk-fr is best. I cannot help further.

**Comment by Yuvrajaryans** [comment]
So now what to do next. Plz help...

**Comment by JonnyTech** [comment]
If XDA or maintainer cannot help then try [Rashr](https://github.com/DsLNeXuS/Rashr/releases/download/2.4.2/RashrApp-release.apk) I was just able to successfully flash my recovery using it, but no guarantees that it will work for you.

**Comment by nailyk-fr** [comment]
@Yuvrajaryans 
> So now what to do next. Plz help...

As my crystal ball ball is broken I cannot guess for you. Some random search: 
https://twrp.me/devices/sonyxperiaz.html
https://forum.xda-developers.com/xperia-z/orig-development/twrp-t3596045

**Read** **all** the instructions on howto install your rom. 
Or ask your ROM maintainer. 

And please close. This is not a TWRP issue. 
Unsubscribed. 

cheers.
